examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,309 @@
+    if (kolmo > 0.99) {
+        return kTRUE;
+    }
+
+    return kFALSE;
+}
+
+std::vector<std::pair<TString, TString>> GetLeafNames(TTree* cbmsim)
+{
+
+    std::vector<TString> ListOfLeaves;
+
+    if (cbmsim) {
+        TObjArray* bla1 = cbmsim->GetListOfLeaves();
+        TIter myiter1(bla1);
+        TBranch* branch;
+        while ((branch = (TBranch*)myiter1.Next())) {
+            TString mystring = branch->GetName();
+            //      cout << "Branch Name: " << mystring << endl;
+
+            // Generate leaf names for points, digis and hits
+            if (mystring.Contains("Point") || mystring.Contains("Digi") || mystring.Contains("Hit")) {
+                TObjArray* bla2 = mystring.Tokenize(".");
+                if (bla2->GetEntriesFast() == 4) {
+                    TString _branch = ((TObjString*)(bla2->At(2)))->GetString();
+                    TString _leaf = ((TObjString*)(bla2->At(3)))->GetString();
+                    if (_leaf.EqualTo("fLink")) {
+                        TString name = _branch + "." + _leaf + ".fLinks";
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fLinks.fFile");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fLinks.fType");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fLinks.fEntry");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fLinks.fIndex");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fLinks.fWeight");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fEntryNr.fFile");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fEntryNr.fType");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fEntryNr.fEntry");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fEntryNr.fIndex");
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf + ".fEntryNr.fWeight");
+                    } else {
+                        ListOfLeaves.emplace_back(_branch + "." + _leaf);
+                    }
+                }
+            }
+
+            if (mystring.Contains("MCTrack")) {
+                TObjArray* bla2 = mystring.Tokenize(".");
+                if (bla2->GetEntriesFast() == 4) {
+                    TString _branch = ((TObjString*)(bla2->At(2)))->GetString();
+                    TString _leaf = ((TObjString*)(bla2->At(3)))->GetString();
+                    ListOfLeaves.emplace_back(_branch + "." + _leaf);
+                }
+            }
+
+            if (mystring.Contains("GeoTracks")) {
+                continue;
+            }
+        }
+    }
+
+    std::vector<std::pair<TString, TString>> leaves;
+    for (auto const element : ListOfLeaves) {
+        TString nameTree2 = "tree2." + element;
+        leaves.emplace_back(make_pair(element, nameTree2));
+    }
```

Comment:
This logic does not fit the abstraction level of this function. See also my [comment at the function signature](https://github.com/FairRootGroup/FairRoot/pull/1419/files#r1247860985). How about moving this out of the function just after the call?

Reply:
Please make a const variable for `"tree2"` and use it here and also [in line 68](https://github.com/FairRootGroup/FairRoot/pull/1419/files/4d8743c0ab0d89991abf8d45f633a2fc51e2e020#diff-67a152a49632b90f589b89a3b5439390f14ea2be3fad4d409e6b8dc364ddb731R68).